build/ctags.go

```diff
@@ -65,10 +67,16 @@ func ctagsAddSymbolsParserMap(todo []*zoekt.Document, languageMap ctags.Language
 			parserKind = ctags.UniversalCTags
 		}
 
-		parser := parserMap[parserKind]
+		parser := parsers[parserKind]
 		if parser == nil {
-			// this happens if CTagsMustSucceed is false and we didn't find the binary
-			continue
+			// Spin up a new parser for this parser kind
+			parser = parserFactory.NewParser(parserKind)
+			if parser == nil {
+				// this happens if CTagsMustSucceed is false and we didn't find the binary
+				continue
+			}
+			parsers[parserKind] = parser
+			defer parser.Close()
```

Comment:
This is a bit messy. In a follow up I plan to:
* Clean up `lockedParser`, overall simplify our parser wrapper
* Pull this multiplexing logic into our parser wrapper, so it just delegates to the right sub-parser process